src/commands/NewIntegrationFileCommand.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as kamel from '../kamel';
+
+const validFilename = require('valid-filename');
+
+const LANGUAGES_WITH_FILENAME_EXTENSIONS = new Map([
+	['Java', 'java'],
+	['XML', 'xml'],
+	['Yaml', 'yaml'],
+	['Groovy', 'groovy'],
+	['JavaScript', 'js'],
+	['Kotlin', 'kts']]);
+const LANGUAGES = Array.from(LANGUAGES_WITH_FILENAME_EXTENSIONS.keys());
+
+export async function create() {
+	const language = await vscode.window.showQuickPick(LANGUAGES);
+	if (language) {
+		const workspaceFolder = await vscode.window.showWorkspaceFolderPick();
+		if (workspaceFolder) {
+			const filename = await vscode.window.showInputBox({
+				prompt: 'Please provide a name for the new file (without extension)',
+				validateInput: (name: string) => {
+					return validateFileName(name, language, workspaceFolder);
+				}
+			});
+			if (filename) {
+				let kamelExe = kamel.create();
+				let newFileFullPath: string = computeFullpath(language, workspaceFolder, filename);
+				kamelExe.invoke(`init ${newFileFullPath}`);
+			}
+		}
+	}
+}
+
+function computeFullpath(language: string, workspaceFolder: vscode.WorkspaceFolder, filename: string) {
```

Comment:
How about defining the return type `string` for the function?

Reply:
I really hate this refactor support in VS Code which is not adding the full signature